### PR TITLE
Added multi-profile support (fixes #4)

### LINKED ISSUE #4: AWS Credentials Multi-Profile Support
Hi all, I'm currently running Caddy from a docker image (latest), and alpine builder (latest)..

I was able to get *Automatic HTTPS* working (thru DNS Validation) by building caddy with route53 dns module and adding the default "Profile" in the AWS Credentials configuration in the `~/.aws/credentials` path as in:

```
[default]
aws_access_key_id = <YOUR_DEFAULT_ACCESS_KEY_ID>
aws_secret_access_key = <YOUR_DEFAULT_SECRET_ACCESS_KEY>
```

---

Everything was working fine until the time I needed to add another "Profile" from another AWS account as in:
```
[development]
aws_access_key_id = <YOUR_DEFAULT_ACCESS_KEY_ID>
aws_secret_access_key = <YOUR_DEFAULT_SECRET_ACCESS_KEY>

[production]
aws_access_key_id = <YOUR_DEFAULT_ACCESS_KEY_ID>
aws_secret_access_key = <YOUR_DEFAULT_SECRET_ACCESS_KEY>
```

And now I don't know how to specify in the virtual-host section of the caddyfile about which aws profile/account it should use to connect for route53 dns validation...

---
Here's my current caddyfile:
````
dev.example.com:443 {
  bind {$ADDRESS}
  header X-Robots-Tag noindex
  reverse_proxy /* dev:80 {
      header_up Host "dev.example.com"
  }
  tls {
      dns route53 # from the development aws account
  }
}
````

````
prod.example.com:443 {
  bind {$ADDRESS}
  header X-Robots-Tag noindex
  reverse_proxy /* dev:80 {
      header_up Host "prod.example.com"
  }
  tls {
      dns route53 # from the production aws account
  }
}
````

---

In contrast to the cloudflare DNS module, I can specify the API key in the caddyfile itself so no worries on how many cloudflare accounts I use it with as in:
````
dev.example.com:443 {
  bind {$ADDRESS}
  header X-Robots-Tag noindex
  reverse_proxy /* dev:80 {
      header_up Host "dev.example.com"
  }
  tls {
      dns cloudflare "redacted-key"
  }
}
````

## **Is there somehow a way to specify the AWS credentials profile that I want to use for DNS validation?**

### PR DESCRIPTION
**Pull request by PhoenixPeca**
Dear caddy team, this PR adds a new argument in the route53 module where people can specify which AWS profile they want to use per virtualhost.. If not defined, it will fall back to the default profile or the profile set in the system environment..

---

Example:

```
  tls {
      dns route53 {
        max_retries 10
        aws_profile "production-profile" # New argument for multi-profile support
      }
  }
```

For details please read this forum thread: https://caddy.community/t/automatic-https-for-route53-multi-profile-support/10512/8

### CODE REVIEW

**Comment by PhoenixPeca** [comment]
I am not sure how you guys handle version releases of this module but i already updated some go modules as basis.. please double check before merging.. i don't want to break other people's usage.. :D

**Review by mholt — CHANGES_REQUESTED** [review]
Definitely lose all the changes in go.mod and go.sum. The rest LGTM.

File: README.md [hunk 1]
```diff
@@ -41,7 +42,8 @@ or with the Caddyfile:
 ```
 tls {
   dns route53 {
-    max_retries 10
+    max_retries 10,
```
Comment (mholt) [inline README.md:45]:
```suggestion
    max_retries 10
```

File: route53.go [hunk 2]
```diff
@@ -53,6 +53,14 @@ func (p *Provider) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 				if d.NextArg() {
 					return d.ArgErr()
 				}
+			case "aws_profile":
+				if d.NextArg() {
+					p.Provider.AwsProfile = repl.ReplaceAll(d.Val(), "")
+					fmt.Println(p.Provider.AwsProfile)
```
Comment (mholt) [inline route53.go:59]:
```suggestion
```

Remove debug line

**Review by mholt — CHANGES_REQUESTED** [review]
I missed a couple things

**Comment by PhoenixPeca** [comment]
Hi @mholt changes have been resolved. Thank you for taking the time to check!. :D

**Review by mholt — APPROVED** [review]

**Comment by danlsgiga** [comment]
LGTM too! Thanks for this PR @PhoenixPeca!

**Pull request merged**